kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PurchaseOrderAmendmentDocument.java

```diff
@@ -1,42 +1,128 @@
-	}
-    
     public String getRouteCode() {
-    	if (StringUtils.isBlank(routeCode)) {
-	    	try{
-	    		BuildingExtension be = (BuildingExtension)(buildingObj.getExtension());
-	    		routeCode = be.getRouteCode();
-	    	}catch(Exception e){
-	    		LOG.debug("Routing Code was not Retrieved");
-	    	}
-    	}
-    	
+        if (StringUtils.isBlank(routeCode)) {
+            try {
+                BuildingExtension be = (BuildingExtension) (buildingObj.getExtension());
+                routeCode = be.getRouteCode();
+            } catch (Exception e) {
+                LOG.debug("Routing Code was not Retrieved");
+            }
+        }
+
         return routeCode;
     }
 
     public void setRouteCode(String routeCode) {
         this.routeCode = routeCode;
     }
-    
+
+    @Override
+    public boolean answerSplitNodeQuestion(String nodeName) throws UnsupportedOperationException {
+        if (nodeName.equals(PurapWorkflowConstants.AMOUNT_REQUIRES_SEPARATION_OF_DUTIES_REVIEW_SPLIT)) {
+            boolean isSeparationOfDutiesReviewRequired = isSeparationOfDutiesReviewRequired();
+            LOG.debug("isSeparationOfDutiesReviewRequired=" + isSeparationOfDutiesReviewRequired);
+            return isSeparationOfDutiesReviewRequired;
+        }
+        if (nodeName.equals(PurapWorkflowConstants.CONTRACT_MANAGEMENT_REVIEW_REQUIRED)) {
+            boolean isContractManagementReviewRequired = isContractManagementReviewRequired();
+            LOG.debug("isSeparationOfDutiesReviewRequired=" + isContractManagementReviewRequired);
+            return isContractManagementReviewRequired;
+        }
+        boolean superAnswerSplitNodeQuestion = super.answerSplitNodeQuestion(nodeName);
+        return superAnswerSplitNodeQuestion;
+    }
+
+    public Set<Person> getAllPriorApprovers() throws WorkflowException {
+        PersonService personService = KimApiServiceLocator.getPersonService();
+        List<ActionTaken> actionsTaken = this.getFinancialSystemDocumentHeader().getWorkflowDocument().getActionsTaken();
+        Set<String> principalIds = new HashSet<String>();
+        Set<Person> persons = new HashSet<Person>();
+
+        for (ActionTaken actionTaken : actionsTaken) {
+            if (KewApiConstants.ACTION_TAKEN_APPROVED_CD.equals(actionTaken.getActionTaken().getCode())) {
+                String principalId = actionTaken.getPrincipalId();
+                if (!principalIds.contains(principalId)) {
+                    principalIds.add(principalId);
+                    persons.add(personService.getPerson(principalId));
+                }
+            }
+        }
+        return persons;
+    }
+
+    protected boolean isSeparationOfDutiesReviewRequired() {
+        try {
+            Set<Person> priorApprovers = this.getAllPriorApprovers();
+
+            // If there are more than 1 prior approvers there is no need for SOD.
+            // If 1 approver exists, check that approver is not the initiator.
```

Comment:
I am really glad this comment is here, since deep nesting is usually a red flag.